spatial/r3/vector_test.go

```diff
@@ -259,17 +259,27 @@ func TestRotate(t *testing.T) {
 var vectorFields = []struct {
 	field      func(Vec) Vec
 	divergence func(Vec) float64
+	jacobian   func(Vec) *Mat
 }{
 	{
 		field: func(v Vec) Vec {
+			// (x*y*z, y*z, z*x)
 			return Vec{X: v.X * v.Y * v.Z, Y: v.Y * v.Z, Z: v.Z * v.X}
 		},
 		divergence: func(v Vec) float64 {
 			return v.X + v.Y*v.Z + v.Z
 		},
+		jacobian: func(v Vec) *Mat {
+			return NewMat([]float64{
+				v.Y * v.Z, v.X * v.Z, v.X * v.Y,
+				0, v.Z, v.Y,
+				v.Z, 0, v.X,
+			})
+		},
 	},
 	{
 		field: func(v Vec) Vec {
+			// (x*y*z*cos(y), y*z+sin(x), z*x/sin(y))
```

Comment:
I think it's better to use continuous functions for testing to avoid false negatives. The test points are generated randomly and it could happen that `y` is close to a multiple of π.